airflow/providers/amazon/aws/transfers/exasol_to_s3.py

```diff
@@ -0,0 +1,117 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Transfers data from Exasol database into a S3 Bucket."""
+
+from tempfile import NamedTemporaryFile
+from typing import Dict, Optional
+
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from airflow.providers.exasol.hooks.exasol import ExasolHook
+from airflow.utils.decorators import apply_defaults
+
+
+class ExasolToS3Operator(BaseOperator):
+    """
+    Export data from Exasol database to AWS S3 bucket.
+
+    :param query_or_table: the sql statement to be executed or table name to export
+    :type query_or_table: str
+    :param key: S3 key that will point to the file
+    :type key: str
+    :param bucket_name: Name of the bucket in which to store the file
+    :type bucket_name: str
+    :param replace: A flag to decide whether or not to overwrite the key
+        if it already exists. If replace is False and the key exists, an
+        error will be raised.
+    :type replace: bool
+    :param encrypt: If True, the file will be encrypted on the server-side
+        by S3 and will be stored in an encrypted form while at rest in S3.
+    :type encrypt: bool
+    :param gzip: If True, the file will be compressed locally
+    :type gzip: bool
+    :param acl_policy: String specifying the canned ACL policy for the file being
+        uploaded to the S3 bucket.
+    :type acl_policy: str
+    :param query_params: Query parameters passed to underlying ``export_to_file``
+        method of ``ExaConnection``.
```

Comment:
Given `ExaConnection` is a concept only existing in the underlying dependency and not in Airflow,  it took me a few moment to figure out what's this "ExaConnection".

Maybe good to provide a little bit more information about `ExaConnection`, like changing this to "**Query parameters passed to underlying ``export_to_file`` method of ``pyexasol``'s ``ExaConnection``.**"

Reply:
Good point, I will add full path for the class

Reply:
Let me know if d76c99d look resonbale

Reply:
LGTM